reverse_proxy_handler.cc

```diff
@@ -0,0 +1,51 @@
+#include "reverse_proxy_handler.h"
+
+ReverseProxyHandler::ReverseProxyHandler() {}
+
+ReverseProxyHandler::~ReverseProxyHandler() {}
+
+ReverseProxyHandler::Status ReverseProxyHandler::Init(const std::string& uri_prefix, const NginxConfig& config) {
+    uri_prefix_ = uri_prefix;
+
+    bool remote_host_found = false;
+    bool remote_port_found = false;
+
+    // expecting two statements in child block: remote_host <remote_host>;
+    //                                          remote_port <remote_port>;
+    if (config.statements_.size() == 2) {
+        for (std::size_t i = 0; i < config.statements_.size(); i++) {
+            if (config.statements_[i]->tokens_.size() != 2) {
```

Comment:
After looking at the original config_parser.h code, it seems like the tokens_ vector will at least be empty and not null such that we can check the size always. It should be all good without needing an extra null pointer check 😄 
Here is the snippet:
class NginxConfigStatement {
 public:
    ...
    std::vector<std::string> tokens_; // Atleast empty vector
}